### PR TITLE
Various fixes and new functions for glib to work

### PR DESCRIPTION
**Pull request by Dennisbonke**
This is a work in progress PR with all the changes needed for glib to compile, I'll update this later when more work is done

### CODE REVIEW

File: options/posix/include/pwd.h [hunk 1]
```diff
@@ -16,6 +16,7 @@ struct passwd {
 	gid_t pw_gid;
 	char *pw_dir;
 	char *pw_shell;
+	char *pw_gecos;
```
Comment (avdgrinten) [inline options/posix/include/pwd.h:19]: Don't we need to adjust the function that parses `/etc/passwd`?

File: options/posix/include/pthread.h [hunk 2]
```diff
@@ -93,6 +93,18 @@ struct  __mlibc_barrier {
 };
 typedef struct __mlibc_barrier pthread_barrier_t;
 
+struct __mlibc_fair_rwlock {
+	pthread_mutex_t m;
+	pthread_mutex_t rw;
```
Comment (avdgrinten) [inline options/posix/include/pthread.h:98]: I think we can get away with declaring these members as `unsigned int`. While we do indeed want two mutexes, we do not need the more complex features that pthread_mutex_t supports.

File: options/posix/include/pthread.h [hunk 3]
```diff
@@ -93,6 +93,18 @@ struct  __mlibc_barrier {
...
+	unsigned read_count;
+};
+typedef struct __mlibc_fair_rwlock pthread_rwlock_t;
+
+struct __mlibc_rwlockattr {
+	int align;
```
Comment (avdgrinten) [inline options/posix/include/pthread.h:104]: Where is this inspired from? musl?

File: options/linux/include/mntent.h [hunk 4]
```diff
@@ -6,6 +6,14 @@
 // TODO: Refer to _PATH_MOUNTED
 #define MOUNTED "/etc/mtab"
 
+/* Generic mount options */
+#define MNTOPT_DEFAULTS "defaults"		/* Use all default options. */
+#define MNTOPT_RO		"ro"			/* Read only. */
+#define MNTOPT_RW		"rw"			/* Read/write. */
+#define MNTOPT_SUID		"suid"			/* Set uid allowed. */
+#define MNTOPT_NOSUID	"nosuid"		/* No set uid allowed. */
+#define MNTOPT_NOAUTO	"noauto"		/* Do not auto mount. */
```
Comment (avdgrinten) [inline options/linux/include/mntent.h:15]: Indentation is a bit broken (maybe use spaces to align this).

File: abis/mlibc/in.h [hunk 5]
```diff
@@ -19,6 +19,7 @@ struct sockaddr_in {
 	struct in_addr sin_addr;
 	uint8_t pad[8];
 };
+#define sin_zero pad		/* for BSD Unix compatibility */
```
Comment (avdgrinten) [inline abis/mlibc/in.h:22]: How do applications use this? (For example, do they care about the size of this field?)

Comment (Dennisbonke) [reply abis/mlibc/in.h:22]: they don't, probably, but iirc you should zero it, which is what `glib` is doing, however, they call it via sin_zero, linux solves that issue by defining it

File: options/posix/include/pthread.h (hunk 3 above)
Comment (Dennisbonke) [reply options/posix/include/pthread.h:104]: the `__mlibc_rwlockattr`? That was modified from linux iirc

File: options/posix/include/pthread.h (hunk 2 above)
Comment (Dennisbonke) [reply options/posix/include/pthread.h:98]: I'll modify that

File: options/posix/include/pwd.h (hunk 1 above)
Comment (Dennisbonke) [reply options/posix/include/pwd.h:19]: Yes we probably should, I'll add that to the list of things this PR still needs

Comment (Dennisbonke) [reply options/posix/include/pwd.h:19]: This should be fixed in 6e2b115

File: options/linux/include/mntent.h (hunk 4 above)
Comment (Dennisbonke) [reply options/linux/include/mntent.h]: Fixed in 9443e93

**Pull request merged**